threedigrid_builder/grid/lines_1d2d.py

```diff
@@ -281,25 +282,20 @@ def assign_2d_node(self, cell_tree: shapely.STRtree) -> None:
         self.line[line_idx, 0] = cell_idx
         self.line_coords[:] = np.nan
 
-    def remove_unassigned(self, nodes) -> "Lines1D2D":
-        """Removes 1D-2D lines where any of the required nodes is set to null, represented as -9999
+    def check_unassigned(self, nodes) -> "Lines1D2D":
+        """Checks 1D-2D lines where any of the required nodes is set to null, represented as -9999
         This is the case when the nodes are outside the 2D domain.
         """
-        rows_removed = self.line[:, 0] == -9999
-        node_ids_removed = self.line[rows_removed, 1]
-        if len(node_ids_removed) > 0:
-            nodes_removed = nodes.id_to_index(node_ids_removed)
-            nodes_removed_formatted = nodes.format_message(nodes_removed)
-
-            logger.warning(
-                f"Removing {len(node_ids_removed)} 1D-2D lines attached to {nodes_removed_formatted} because they are outside of the DEM."
+        invalid_rows = self.line[:, 0] == -9999
+        invalid_node_ids = self.line[invalid_rows, 1]
+        if len(invalid_node_ids) > 0:
+            invalid_nodes = nodes.id_to_index(invalid_node_ids)
+            invalid_nodes_formatted = nodes.format_message(invalid_nodes)
+
+            raise SchematisationError(
+                f"The following {invalid_nodes_formatted} are outside of the DEM, but are connected to an object which does not have an ISOLATED calculation type."
```

Comment:
- Technically it is not "outside of the DEM" but "outside of the 2D model domain"; the 2D model domain is in general slightly larger than the DEM because.


```suggestion
                f"The following objects have a (double) connected calculation type, but are outside of the 2D model domain: {invalid_nodes_formatted}."
```